contactsummary/tests/phpunit/CRM/Contactsummary/Utils/ContractSummaryTestTrait.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+/**
+ * A trait with helper methods to be reused among this extension's tests
+ */
+trait ContractSummaryTestTrait {
+
+  /**
+   * Creates absence period from specified start and end dates and return its ID
+   *
+   * @param $startDate
+   * @param $endDate
+   *
+   * @return int
+   */
+  protected function createAbsencePeriod($startDate, $endDate)  {
+    $params = array('name'=> $startDate,'start_date'=>$startDate, 'end_date'=>$endDate);
+    $period = civicrm_api3('HRAbsencePeriod', 'create', $params);
+    return $period['id'];
+  }
+
+  /**
+   * Creates absence entitlements for specific leave type
+   *
+   * @param $params
+   *
+   * @return int
+   */
+  protected function createAbsenceEntitlement($params)  {
+    $period = CRM_HRAbsence_BAO_HRAbsenceEntitlement::create($params);
```

Comment:
That's because of the poor way Civi handles dates. When you're using the BAO, you need to pass the dates for it in 'YmdHis' format. The API worked because it automatically converts the date to this format. You can find more details here: https://compucorp.atlassian.net/wiki/display/PCHR/Handling+dates+in+CiviCRM